feat(ngRoute):change pathRegExp to respect optional wildcard route params

murphomatic opened the pull request:
change pathRegExp to accept the two-character option of '*?' on route params to allow them to be wildcard and optional at the same time.  Motivation for change inspired by the need to accept an auth token on the URL that contained slashes (need for wildcard param) but this token on the URL was not required (need for optional param).  Previous behavior only saw the param as wildcard and not optional, which resulted in the need for a trailing slash on the url (e.g. /login/ vs. /login)

new feature - does not address any open issues

Commit:
feat(ngRoute):change pathRegExp to respect optional wildcard route params

change pathRegExp to accept the two-character option of '*?' on route params to allow them to be wildcard and optional at the same time.  Motivation for change inspired by the need to accept an auth token on the URL that contained slashes (need for wildcard param) but this token on the URL was not required (need for optional param).  Previous behavior only saw the param as wildcard and not optional, which resulted in the need for a trailing slash on the url (e.g. /login/ vs. /login)

new feature - does not address any open issues

gkalpak commented on package.json:
```diff
@@ -32,13 +32,13 @@
     "grunt-contrib-connect": "~0.8.0",
     "grunt-contrib-copy": "~0.6.0",
-    "grunt-contrib-jshint": "~0.10.0",
+    "grunt-contrib-jshint": "https://registry.npmjs.org/grunt-contrib-jshint/-/grunt-contrib-jshint-0.10.0.tgz",
```
These changes should be removed.

gkalpak commented on test/ngRoute/routeSpec.js:
```diff
@@ -974,6 +974,32 @@ describe('$route', function() {
+
+      inject(function($location, $rootScope, $route) {
+        $location.path('/foo/bar1/bar2/bar3/baz');
+	$rootScope.$digest();
```
Indentation is off.

gkalpak commented:
LGTM with the comments addressed :+1:

Commit:
feat(ngRoute):change pathRegExp to respect optional wildcard route params

change pathRegExp to accept the two-character option of '*?' on route params to allow them to be wildcard and optional at the same time.  Motivation for change inspired by the need to accept an auth token on the URL that contained slashes (need for wildcard param) but this token on the URL was not required (need for optional param).  Previous behavior only saw the param as wildcard and not optional, which resulted in the need for a trailing slash on the url (e.g. /login/ vs. /login)

new feature - does not address any open issues

Commit:
feat(ngRoute):change pathRegExp to respect optional wildcard route params

change pathRegExp to accept the two-character option of '*?' on route params to allow them to be wildcard and optional at the same time.  Motivation for change inspired by the need to accept an auth token on the URL that contained slashes (need for wildcard param) but this token on the URL was not required (need for optional param).  Previous behavior only saw the param as wildcard and not optional, which resulted in the need for a trailing slash on the url (e.g. /login/ vs. /login)

new feature - does not address any open issues

murphomatic commented:
apologies for the miss, everything should be set now.

murphomatic commented:
@gkalpak Let me know if there is any other work you see that needs to be done (I believe I've satisfied the issues you've commented on).  Thank you!

gkalpak commented:
Merged ! Thx @murphomatic :+1: 
Backported to `v1.5.x` (c0797c6) and `v1.4.x` (6a4403a).

murphomatic commented:
Awesome! Thanks @gkalpak!
